src/operations/filter.ts

```diff
@@ -10,12 +10,13 @@ function defaultFilteringFunction<T>(item: T, filteringText: string, filteringFi
   filteringFields = filteringFields || Object.keys(item as Record<string, any>);
   const lowFilteringText = filteringText.toLowerCase();
 
-  return filteringFields.some(
-    key =>
-      String((item as Record<string, any>)[key])
-        .toLowerCase()
-        .indexOf(lowFilteringText) > -1
-  );
+  return filteringFields.some(key => {
+    const value = (item as Record<string, any>)[key];
+    if (value && typeof value === 'object') {
```

Comment:
With this condition, the item `{key: undefined}` will pass `{filteringText: 'undefined'}` filter. Same for `null`.

Because React renders `null` and `undefined` as empty, this filtering might be unexpected

Reply:
we have some explicit tests for this, so I assume that it was intentional behavior that should be kept: https://github.com/cloudscape-design/collection-hooks/blob/7af5676e47a404d25e387d9cdeca3d055c767fed/src/__tests__/operations/filter.test.ts#L82-L92

Reply:
Looks like a missed opportunity when we released v3.0, but okay. This fix is already very good.